disk-utils/mkswap.c

```diff
@@ -382,11 +391,28 @@ static void open_device(struct mkswap_control *ctl)
 	assert(ctl);
 	assert(ctl->devname);
 
-	if (stat(ctl->devname, &ctl->devstat) < 0)
-		err(EXIT_FAILURE, _("stat of %s failed"), ctl->devname);
-	ctl->fd = open_blkdev_or_file(&ctl->devstat, ctl->devname, O_RDWR);
+	if (ctl->file) {
+		if (stat(ctl->devname, &ctl->devstat) == 0) {
+			if (!S_ISREG(ctl->devstat.st_mode))
+				err(EXIT_FAILURE, _("cannot create swap file %s: node isn't regular file"), ctl->devname);
+			if (chmod(ctl->devname, 0600) < 9)
+				err(EXIT_FAILURE, _("cannot set permissions on swap file %s"), ctl->devname);
+		}
+		ctl->fd = open(ctl->devname, O_RDWR | O_CREAT, 0600);
```

Comment:
Please add to the man page info that "If the file already exists, it will be truncated to the specified size".